src/pages/WebMeetWaitingRoom.js

```diff
@@ -1,17 +1,152 @@
-import React, { useEffect, useState, useMemo } from "react";
+/*global JitsiMeetExternalAPI */
+import React, {
+  useEffect,
+  useState,
+  useMemo,
+  useRef,
+  useCallback,
+} from "react";
 import "./home.css";
 import * as patientService from "../services/patient";
 import { getDateString, getTimeString, getToday } from "../utils";
 import { BOOKING_STATUS } from "../constants/constants";
+import config from "../config/index";
+import Pusher from "pusher-js";
 
 const WebMeetWaitingRoom = ({ patientDetails }) => {
   const [appointment, setAppointment] = useState({});
+  const jitsiContainer = useRef(null);
+  const [jitsiApi, setJitsiApi] = useState(null);
+  const [startWebMeeting, setStartWebMeeting] = useState(false);
+
   const today = getToday();
   useEffect(() => {
     window.scrollTo(0, 0);
     getAppointmentDetail();
   }, []);
 
+  const handleAppointmentProgress = useCallback(
+    (data) => {
+      setAppointment({ ...appointment, eventStatusDesc: "InProgress" });
+
+      if (data) {
+        setStartWebMeeting(true);
+      }
+    },
+    [appointment]
+  );
+
+  const handleAppointmentCompleted = useCallback(
+    (data) => {
+      setAppointment({ ...appointment, eventStatusDesc: "Completed" });
+
+      if (data) {
+        setStartWebMeeting(false);
+      }
+    },
+    [appointment]
+  );
+
+  useEffect(() => {
+    const { key, cluster } = config.pusher;
+    const pusher = new Pusher(key, {
+      cluster,
+      encrypted: true,
+    });
+    const appointmentStatus = pusher.subscribe("appointmentStatus");
+
+    if (appointment.organizationEventBookingId && !startWebMeeting) {
+
+      appointmentStatus.bind("InProgress", handleAppointmentProgress);
+    }
+
+    if (appointment.organizationEventBookingId) {
+      appointmentStatus.bind("completed", handleAppointmentCompleted);
+    }
+  }, [appointment, startWebMeeting]);
+
```

Comment:
this is making multiple connections to pusher. we need to change this so that it only makes the connection once

Reply:
resolved